core/src/avm2/activation.rs

```diff
@@ -3025,7 +3027,8 @@ impl<'a, 'gc> Activation<'a, 'gc> {
 
         let address =
             usize::try_from(address).map_err(|_| "RangeError: The specified range is invalid")?;
```

Comment:
This should be `RangeError: Error #1506: The specified range is invalid.` (for negative addresses)

Reply:
I haven't touched those errors in this PR, can we do that in a followup?